nft_ingester/src/metadata_workers/persister.rs

```diff
@@ -0,0 +1,82 @@
+use std::sync::Arc;
+
+use interface::{
+    error::JsonDownloaderError,
+    json_metadata::{JsonPersister, MetadataDownloadResult},
+};
+use tokio::{
+    sync::mpsc::Receiver,
+    time::{sleep, Duration},
+};
+use tracing::{debug, error};
+
+pub const JSON_BATCH: usize = 300;
+pub const WIPE_PERIOD_SECS: u64 = 60;
+pub const SLEEP_TIME_SECS: u64 = 1;
+
+pub struct TasksPersister<T: JsonPersister + Send + Sync + 'static> {
+    pub persister: Arc<T>,
+    pub json_receiver: Receiver<(String, Result<MetadataDownloadResult, JsonDownloaderError>)>,
+    pub shutdown_rx: tokio::sync::broadcast::Receiver<()>,
+}
+
+impl<T: JsonPersister + Send + Sync + 'static> TasksPersister<T> {
+    pub fn new(
+        persister: Arc<T>,
+        json_receiver: Receiver<(String, Result<MetadataDownloadResult, JsonDownloaderError>)>,
+        shutdown_rx: tokio::sync::broadcast::Receiver<()>,
+    ) -> Self {
+        Self { persister, json_receiver, shutdown_rx }
+    }
+
+    pub async fn run(mut self) {
+        let mut buffer = vec![];
+        let persister = self.persister.clone();
+
+        tokio::select! {
+            _ = self.shutdown_rx.recv() => {
+                if let Err(e) = self.persister.persist_response(buffer).await {
+                    error!("Could not save JSONs to the storage: {:?}", e);
+                } else {
+                    debug!("Saved metadata successfully...");
+                }
+            }
+            _ = Self::process_persisting(persister, self.json_receiver, &mut buffer) => {}
+        }
+    }
+
+    async fn process_persisting(
+        persister: Arc<T>,
+        mut json_receiver: Receiver<(String, Result<MetadataDownloadResult, JsonDownloaderError>)>,
+        buffer: &mut Vec<(String, Result<MetadataDownloadResult, JsonDownloaderError>)>,
+    ) {
+        loop {
```

Comment:
I know, this approach of buffering is not new and is here for ages. wdyt of this whole approach? Do you consider its added complexity adds value?

Reply:
Actually, I do not see how it can be implemented with batch updates and without a buffer.